File: src/main/java/com/skylotoneoneone/service/SubscriptionService.java
package com.skylotoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skylotoneoneone.domain.Subscription;
import com.skylotoneoneone.dto.SubscriptionDTO;
import com.skylotoneoneone.dto.SubscriptionSearchDTO;
import com.skylotoneoneone.dto.SubscriptionPageDTO;
import com.skylotoneoneone.dto.SubscriptionConvertCriteriaDTO;
import com.skylotoneoneone.service.GenericService;
import com.skylotoneoneone.dto.common.RequestDTO;
import com.skylotoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SubscriptionService extends GenericService<Subscription, Integer> {

	List<Subscription> findAll();

	ResultDTO addSubscription(SubscriptionDTO subscriptionDTO, RequestDTO requestDTO);

	ResultDTO updateSubscription(SubscriptionDTO subscriptionDTO, RequestDTO requestDTO);

    Page<Subscription> getAllSubscriptions(Pageable pageable);

    Page<Subscription> getAllSubscriptions(Specification<Subscription> spec, Pageable pageable);

	ResponseEntity<SubscriptionPageDTO> getSubscriptions(SubscriptionSearchDTO subscriptionSearchDTO);
	
	List<SubscriptionDTO> convertSubscriptionsToSubscriptionDTOs(List<Subscription> subscriptions, SubscriptionConvertCriteriaDTO convertCriteria);

	SubscriptionDTO getSubscriptionDTOById(Integer subscriptionId);







}

File: skylotoneoneone_ui/src/services/SubscriptionService.js
import http from "../http-common"; 

class SubscriptionService {
  getAllSubscriptions(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/subscription/subscriptions`, searchDTO);
  }

  get(subscriptionId) {
    return this.getRequest(`/subscription/${subscriptionId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/subscription?field=${matchData}`, null);
  }

  addSubscription(data) {
    return http.post("/subscription/addSubscription", data);
  }

  update(data) {
  	return http.post("/subscription/updateSubscription", data);
  }
  
  uploadImage(data,subscriptionId) {
  	return http.postForm("/subscription/uploadImage/"+subscriptionId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SubscriptionService();
